src/networking.c

```diff
@@ -2878,6 +2934,47 @@ int clientSetNameOrReply(client *c, robj *name) {
     return result;
 }
 
+/* This function implements CLIENT SETMETA, including replying to the
+ * user with an error if the charset is wrong (in that case C_ERR is
+ * returned). If the function succeeded C_OK is returned, and it's up
+ * to the caller to send a reply if needed.
+ *
+ * Setting an empty string as meta has the effect of unsetting the
+ * currently set meta: the client will remain without meta.*/
+int clientSetMetaOrReply(client *c)
```

Comment:
I don't like that this API behaves slightly differently in the no-argument case vs the multi-argument case. With no arguments, it's delete, whereas with arguments it's basically a merge operation. I think it makes sense to always be an update operation, where it completely replaces the current metadata with new metadata. I would expect clients to generally just set this once on startup, not incrementally build up an object.